stubs/babel/babel/numbers.pyi

```diff
@@ -34,6 +34,7 @@ def get_decimal_quantum(precision): ...
 def format_decimal(
     number, format: Any | None = ..., locale=..., decimal_quantization: bool = ..., group_separator: bool = ...
 ): ...
+def format_compact_decimal(number, *, format_type: str = ..., locale=..., fraction_digits: int = ...): ...
```

Comment:
- Looks like `locale` can be anything that can be passed to `babel.core.Locale.parse`, which means it should be `Locale | str | None`
- Looks like the function returns whatever `babel.NumberPattern.apply` returns, and I think that only ever returns a `str`

```suggestion
def format_compact_decimal(number, *, format_type: str = ..., locale: Locale | str | None =..., fraction_digits: int = ...) -> str: ...
```

Reply:
I had left it without to be consistent with the functions around it as I'm not sure I have the time to fix annotations for the entire library.

I suppose it makes sense to annotate new functions accurately regardless, though.

Reply:
Stubs are generally incomplete when they're first contributed to typeshed, which is fine (it's a lot of work to contribute stubs for a new package!) — but we reserve the right to be slightly pickier for smaller PRs like this, which only touch a few functions ;)